dfs/destroy.go

```diff
@@ -44,14 +39,25 @@ func (dfs *DistributedFilesystem) Destroy(tenantID string) error {
 	if err := dfs.deleteImages(tenantID, docker.Latest); err != nil {
 		return err
 	}
-	if err := dfs.net.Stop(); err != nil {
-		glog.Errorf("Could not stop nfs server: %s", err)
+
+	volumeRemoveFailed := true
+	dfs.net.RemoveVolume(vol.Path())
+	// sync will unbind the the volume from exports dir, making it not busy
+	if err := dfs.net.Sync(); err != nil {
+		glog.Errorf("Could not sync volume destroy for tenant %s: %s", tenantID, err)
 		return err
 	}
-	defer dfs.net.Restart()
+	defer func() {
```

Comment:
no need to defer here--the facade will have already deleted the services from elastic by the time you get to this point.  So even if the disk removal fails, you should still try to remove the binding.